src/pages/ModuleEditor/ModuleEditor.tsx

```diff
@@ -47,12 +53,16 @@ export default function ModuleEditor({match: {params: {moduleId}}}: Props) {
   const onSubmit = async (form: ModuleForm, {setErrors}: FormikHelpers<ModuleForm>) => {
     setMessage(undefined);
     try {
+      await deleteModuleTags(deletedModuleTags);
+      form.moduleTags = moduleTags;
       const response = await saveModule(form);
```

Comment:
Let the backend reconcile which tags need to be deleted instead of putting this load on the frontend.